nbflow/scons.py

```diff
@@ -49,9 +51,12 @@ def print_cmd_line(s, targets, sources, env):
             sys.stdout.write("%s --> %s\n"% (str(sources[0]), str(target)))
 
 
-def setup(env, directories):
+def setup(env, directories, timeout=None):
     env['PRINT_CMD_LINE_FUNC'] = print_cmd_line
     env.Decider('timestamp-newer')
+    if timeout is not None:
```

Comment:
Hmm, it would be nice if we could come up with a way to do this without using global variables (see [here](https://docs.quantifiedcode.com/python-anti-patterns/maintainability/using_the_global_statement.html) for an explanation of why avoiding globals is a good idea).

One way this might be possible is to create a class, say called `Builder`, that has an attribute called `timeout` and that has a method called `build_notebook`. Then here you can create an instance of builder, set the timeout value on it, and then use the method on the instance below in `env.Command`.

Reply:
In the next commit, intead of using classes, I chose a more functional approach and use only pure functions and define a default value for `timeout` in `build_notebook`. This seems more cleaner to me.

Because of the nature of `env.Command` I pass a new function `build_notebook_timeout` which is a partial evaluation of `build_notebook` in case a `timeout` is set or the `build_notebook` function in case no `timeout` is passed.